--- xterm-322/charproc.c	2015-12-30 10:38:49.000000000 +0100
+++ xterm-322.diff/charproc.c	2017-07-12 22:14:07.430896393 +0200
@@ -653,7 +653,7 @@
 #endif				/* OPT_ISO_COLORS */
 
     CLICK_RES("2", screen.onClick[1], "word"),
-    CLICK_RES("3", screen.onClick[2], "line"),
+    CLICK_RES("3", screen.onClick[2], "group"),
     CLICK_RES("4", screen.onClick[3], 0),
     CLICK_RES("5", screen.onClick[4], 0),
 
